receiver/jaegerreceiver/factory.go

```diff
@@ -59,33 +64,42 @@ func (f *Factory) Type() string {
 	return typeStr
 }
 
-// CustomUnmarshaler returns nil because we don't need custom unmarshaling for this config.
+// CustomUnmarshaler is used to add defaults for named but empty protocols
 func (f *Factory) CustomUnmarshaler() receiver.CustomUnmarshaler {
-	return nil
+	return func(v *viper.Viper, viperKey string, intoCfg interface{}) error {
+		err := v.UnmarshalKey(viperKey, intoCfg, func(decoderCfg *mapstructure.DecoderConfig) {})
```

Comment:
This doesn't seem to be an equivalent functionality to what it was before. When there is no customer unmarhaler defined Config uses `UnmarshalExact` which errors if there is an option that does not exist in the struct. `UnmarshalKey` doesn't behave that way.

Reply:
This is true.  I chose this b/c `UnmarshalExact` would not add an entry to the map if there were no subkeys. i.e.
```
jaeger:
  protocols:
    grpc:  
```
would not unmarshal to a "grpc" entry in the `receiverCfg.protocols` map where as `UnmarshalKey` did.  I agree this functionality is different.  I will submit a better implementation.

Reply:
I'm reviewing options to better reproduce the original functionality along with the custom functionality we need.  @tigrannajaryan are you ok with exporting this method:

https://github.com/open-telemetry/opentelemetry-collector/blob/60b03d0d2d503351501291b30836d2126487a741/config/config.go#L857-L859

from the config package.  There is a lot of custom functionality in here related to expanding env vars I'd have to reproduce in my custom unmarshaller to get the same behavior.  I think this functionality could be useful for future custom unmarshallers as well.

Reply:
Yes, I think it is OK to export getConfigSection. It implements a generic functionality that we should be able to provide as a public API for components that need custom unmarshaling. Let's see the implementation you have for customer unmarshaler.